### PR TITLE
Business Registration Endpoint

### PR DESCRIPTION
**Pull request by juliamarco**
### This Pull Request Implements:

- Build Endpoint /api/v1/businesses for creating new businesses in our DB
- Add Test for happy and sad path for business creation
- If the request is successful, it will return back the api_key generated and 201 status
- If the record could not be saved, it will return a 422 status and a message explaining the error
- [x] Linked To User Story on Trello

### CODE REVIEW

**Comment by juliamarco** [comment]
[Business registration endpoint](https://trello.com/c/6yBacrWn/1-business-registration-endpoint)

File: app/controllers/api/v1/businesses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Api::V1::BusinessesController < ApplicationController
+
+  def create
+    business = Business.create_new(business_params, code_params[:distributor_code])
```
Comment (Mackenzie-Frey) [inline app/controllers/api/v1/businesses_controller.rb]: I like how you broke out the create_new method.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,9 @@
 class ApplicationController < ActionController::API
+
+  rescue_from ActiveRecord::RecordInvalid, with: :render_unprocessable_entity_response
+
+  def render_unprocessable_entity_response(exception)
```
Comment (Mackenzie-Frey) [inline app/controllers/application_controller.rb]: Glad you added this.

File: app/models/business.rb [hunk 3]
```diff
@@ -17,4 +17,16 @@ class Business < ApplicationRecord
     only_integer: true,
     greater_than_or_equal_to: 1
   }
+
+  def self.create_new(params, code)
```
Comment (Mackenzie-Frey) [inline app/models/business.rb]: I think we need a model test for this method.

**Pull request merged**